### PR TITLE
Small optimizations using cached VM Script

### PR DESCRIPTION
**Pull request by shargon**
Require https://github.com/neo-project/neo-vm/pull/221 https://github.com/neo-project/neo/pull/1178

### CODE REVIEW

**Comment by erikzhang** [comment]
Please wait for #1178

**Comment by erikzhang** [comment]
@shargon Conflicts.

**Comment by shargon** [comment]
Done!

File: neo/Ledger/Blockchain.cs [hunk 1]
```diff
@@ -100,7 +99,7 @@ public Blockchain(NeoSystem system, Store store)
             this.system = system;
             this.MemPool = new MemoryPool(system, ProtocolSettings.Default.MemoryPoolMaxTransactions);
             this.Store = store;
-            lock (lockObj)
+            lock (typeof(Blockchain))
```
Comment (lock9) [inline neo/Ledger/Blockchain.cs]: This doesn't seem to be recommended:
https://bytes.com/topic/c-sharp/answers/249277-dont-lock-type-objects

I'm not a c# expert, but is this change really needed?

Comment (shargon) [reply neo/Ledger/Blockchain.cs]: We can change it, but one object in memory for only once, seems weird to me

Comment (lock9) [reply neo/Ledger/Blockchain.cs]: The author is very clear about it: `This practice is not acceptable, let alone
recommended.`

Comment (shargon) [reply neo/Ledger/Blockchain.cs]: Done! if is bad is bad :)
